### PR TITLE
Add troubleshooting headless TCP service disconnects.

### PR DESCRIPTION
**Pull request by hollinwilkins**
This PR adds documentation on how to prevent Istio from disconnecting connections to headless TCP services every 15 minutes or so.

See conversation here for reference:
https://github.com/istio/istio/issues/506

### CODE REVIEW

File: _help/troubleshooting.md [hunk 1]
```diff
@@ -325,3 +325,29 @@ Check your `ulimit -a`. Many systems have a 1024 open file descriptor limit by d
 ```
 
 Make sure to raise your ulimit. Example: `ulimit -n 16384`
+
+## Headless TCP Services Losing Connection from Istiofied Containers
+
+Envoy is reset every 15 minutes by the `istio-ca` deployment, as a part
```
Comment (andraxylia) [inline _help/troubleshooting.md]: If `istio-ca` is deployed, Envoy is restarted every 15 minutes to refresh the certificates, causing the disconnection of TCP streams or long-running connections.

Comment (hollinwilkins) [reply _help/troubleshooting.md]: @andraxylia Do you want me to make these changes?

Comment (ldemailly) [reply _help/troubleshooting.md]: yes please do

File: _help/troubleshooting.md [hunk 2]
```diff
@@ -325,3 +325,29 @@ Check your `ulimit -a`. Many systems have a 1024 open file descriptor limit by d
...
+of this process it will restart Envoy sidecars causing them to
+disconnect TCP streams. In order to stop this from happening, you will
```
Comment (ldemailly) [inline _help/troubleshooting.md]: I would insert something like:
"While your TCP services should be resilient to rare connection close events (as your pods may move for a number of reasons instance), if you really need to keep long running connection, and until we release the version of istio that doesn't require envoy restart for certificate reload, you will have to disable mTLS [...]"
(though that's kind of a loooong sentence)

File: _help/troubleshooting.md (hunk 1 above)
Comment (andraxylia) [reply _help/troubleshooting.md]: Yes, please rephrase, sorry if that was not clear. Envoy is not restarted by the istio-ca, but by the pilot agent.

File: _help/troubleshooting.md (hunk 2 above)
Comment (hollinwilkins) [reply _help/troubleshooting.md]: @ldemailly 30 seconds isn't a long TCP connection. DB connections often stay open for hours by default before being recycled. Will add something about connection resilience as well.

**Comment by hollinwilkins** [comment]
Hey, I made the requested changes, had to open a new PR because the old repository was deleted:

https://github.com/istio/istio.github.io/pull/870

**Comment by ldemailly** [comment]
closing -> #870

**Pull request closed without merging**